### PR TITLE
[influxdb-enterprise] Enable configuration of environment variables

### PR DESCRIPTION
**Pull request by brightstone9**
It shoule be possible to configure environment variables in enterprise chart as well. 
There is another PR related to this change (https://github.com/influxdata/helm-charts/pull/290) but didn't seem to be getting the update it was needed. 

- [ ] CHANGELOG.md updated
- [x] Rebased/mergable
- [x] Tests pass
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

**changed files**
* update meta-statefulset.ymal
* update data-statefulset.yaml
* update values.yaml
* update Chart.yaml (chart version to v0.1.17)

**Applied test 1**
`helm lint ./influxdb-enterprise`
**Result**
```
==> Linting ./influxdb-enterprise
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
```

**Applied test 2**
`helm template --set data.env.INFLUXDB_ANTI_ENTROPY_ENABLED=true influxdb-enterprise ./influxdb-enterprise`
**Result : data-statefulset.yaml**
```
# Source: influxdb-enterprise/templates/data-statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: influxdb-enterprise-data
  labels:
    helm.sh/chart: influxdb-enterprise-0.1.17
    app.kubernetes.io/name: influxdb-enterprise
    app.kubernetes.io/instance: influxdb-enterprise
    app.kubernetes.io/version: "1.9.6"
    app.kubernetes.io/managed-by: Helm
spec:
  replicas: 1
  podManagementPolicy: Parallel
  serviceName: influxdb-enterprise-data
  selector:
    matchLabels:
      influxdb.influxdata.com/component: data
      app.kubernetes.io/name: influxdb-enterprise
      app.kubernetes.io/instance: influxdb-enterprise
  template:
    metadata:
      labels:
        influxdb.influxdata.com/component: data
        app.kubernetes.io/name: influxdb-enterprise
        app.kubernetes.io/instance: influxdb-enterprise
    spec:
      securityContext:
        null
      serviceAccountName: default
      volumes:
      - name: influxdb-enterprise-data-data
        emptyDir: {}
      - name: config
        configMap:
          name: influxdb-enterprise-data
      - name: tls
        secret:
          secretName: influxdb-enterprise-data-tls

      containers:
        - name: influxdb-enterprise
          command:
          - "/usr/bin/perl"
          args:
          - "/etc/influxdb/entrypoint.pl"
          securityContext:
            null
          image: "influxdb:1.9.6-data"
          imagePullPolicy:
          env:
          - name: RELEASE_NAME
            value: influxdb-enterprise
          - name: INFLUXDB_ANTI_ENTROPY_ENABLED
            value: "true"
          ports:
            - name: http
              containerPort: 8086
              protocol: TCP
            - name: raft
              containerPort: 8088
              protocol: TCP
            - name: udp
              containerPort: 8089
              protocol: UDP
            - name: graphite
              containerPort: 2003
              protocol: TCP
            - name: opentsdb
              containerPort: 4242
              protocol: TCP
            - name: collectd
              containerPort: 25826
              protocol: UDP
          livenessProbe:
            httpGet:
              path: /ping
              port: http
              scheme: HTTPS
          readinessProbe:
            initialDelaySeconds: 30
            httpGet:
              path: /ping
              port: http
              scheme: HTTPS
```

### CODE REVIEW

File: charts/influxdb-enterprise/templates/data-statefulset.yaml [hunk 1]
```diff
@@ -88,6 +88,10 @@ spec:
           env:
           - name: RELEASE_NAME
             value: {{ include "influxdb-enterprise.fullname" . }}
+          {{- range $key, $val := .Values.data.env }}
+          - name: {{ $key }}
+            value: {{ $val | quote }}
+          {{- end}}
```
Comment (alespour) [inline charts/influxdb-enterprise/templates/data-statefulset.yaml:94]: Other charts use simpler way, which also does not unnecessarily quote values
```suggestion
          {{- if .Values.data.env }}
{{ toYaml .Values.data.env | indent 10 }}
          {{- end}}
```

File: charts/influxdb-enterprise/templates/meta-statefulset.yaml [hunk 2]
```diff
@@ -91,6 +91,10 @@ spec:
                 secretKeyRef:
                   name: {{ .Values.meta.sharedSecret.secretName }}
                   key: secret
+          {{- range $key, $val := .Values.meta.env }}
+            - name: {{ $key }}
+              value: {{ $val | quote }}
+          {{- end}}
```
Comment (alespour) [inline charts/influxdb-enterprise/templates/meta-statefulset.yaml:97]: Same here
```suggestion
          {{- if .Values.meta.env }}
{{ toYaml .Values.meta.env | indent 12 }}
          {{- end}}
```

**Review by alespour — CHANGES_REQUESTED** [review]
Could you rebase and resolve the conflict in `values.yaml`, please.

**Comment by brightstone9** [comment]
I messed up while rebasing I will open a new merge request instead...

**Pull request closed without merging**